### DISCUSSION TITLE
Is the component generator ready?

### DISCUSSION THREAD

**Discussion by CeSun** [Dreamescaper/BlazorBindings.Maui#93] [General]
I noticed that there is a project called "BlazorBindings.Maui.ComponentGenerator" in the warehouse, which seems to be used to adapt MAUI controls, but this package is not referenced in the project template. Is it already available?

**Comment by Dreamescaper** [comment]
It is available, but lacks a proper documentation page (that's the most boring part :) ). I've mentioned it [here](https://dreamescaper.github.io/MobileBlazorBindingsDocs/maui-blazor-bindings/release-notes/preview-9.html#component-generator) with a basic usage example.
Note that it is a dotnet tool, not a regular nuget package, so you don't need to reference it in your project. 
Also, since the project's API is not fully stable (especially "internal" staff), you'll probably need to regenerate your controls when updating to a new version.

Please let me know of your experience and any issues you encounter!

**Reply by CeSun** [reply]
I tried the component generator, the component library is: https://github.com/yiszza/Material.Components.Maui, I found that if the inherited parent component of the component is also a custom component, then the parent component also needs to be added manually [assembly : GenerateComponent(typeof(XX))], it will not automatically recurse and generate.

**Reply by CeSun** [reply]
The generated code has compilation errors. The generator generates two "ContextMenu" properties for the "SKTouchCanvasView" component. Their types are different, but their names are the same.
I try to rename one of the properties to ContextMenu2, and modify the compilation error, and finally the project compiles through, but the operation will crash.

My test project: https://github.com/CeSun/App1

**Reply by Dreamescaper** [reply]
Thanks for a feedback. 
The issue with duplicate properties is already fixed, but not published to nuget so far.
In the meantime, you can simply remove it
![image](https://user-images.githubusercontent.com/17177729/211192396-cc8d1d9a-4b89-4314-9ea7-bd85c3bfa987.png)


Could you clarify about operation crash? Probably it should be fine after you add Material resources (the one you've asked in another discussion).

**Reply by CeSun** [reply]
> Thanks for a feedback. 
> The issue with duplicate properties is already fixed, but not published to nuget so far.
> In the meantime, you can simply remove it
> ![image](https://user-images.githubusercontent.com/17177729/211192396-cc8d1d9a-4b89-4314-9ea7-bd85c3bfa987.png)
> 
> 
> Could you clarify about operation crash? Probably it should be fine after you add Material resources (the one you've asked in another discussion).

It's worked, Thank you😁.
